### PR TITLE
Support config file reloads with useConfigPrecisely

### PR DESCRIPTION
**Pull request by glasser**

### CODE REVIEW

**Comment by glasser** [comment]
Tests now pass, yay.  This adds a CHANGELOG that links to https://github.com/apollographql/engine-docs/pull/81 so please review that too

I'll make sure to review https://github.com/apollographql/apollo-engine-js/pull/144 and get it merged before release, but I have to run now for a bit.

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,16 +12,21 @@ the Engine documenation.
...
+  including changes in `2018.02-84-g7a295e631` and `2018.02-50-gef2fc6d4e`.
 - Simplify how the apollo-engine npm module communicates with the Engine Proxy
   binary.  **Backwards-incompatible changes**:
   + The `logger` option to `new Engine` added in 0.8.9 no longer exists. It is
     replaced by `proxyStdoutStream` and `proxyStderrStream` options, as well as
     a `restarting` event on the `Engine` object.
   + The default log style is now the same as in the Docker container release of
     engineproxy: textual logs over stdout, instead of JSON over stderr.
-- allowFullConfiguration option -- XXX this may need to be improved before release
+- New `allowFullConfiguration` option (for single proxy mode) which uses the
```
Comment (zionts) [inline CHANGELOG.md]: Should be `useConfigPrecisely` option

File: README.md [hunk 2]
```diff
@@ -58,18 +60,22 @@ This is the minimum necessary information in the engine configuration object to
...
+  "startupTimeout": 5000,           // If >0, .start() will throw if the proxy binary does not finish
+                                    // startup within the given number of milliseconds. Defaults to 5000ms.
+  "useConfigPrecisely": true        // For use with single proxy mode. Set this to true for the spawned
+                                    // Engine Proxy process to use the engineConfig object
+                                    // specified in the constructor without automatically setting up an
+                                    // origin and frontend.
 
   // Shortcuts to "origins" in EngineConfig
   "origin": {
     "requestTimeout": "5s",          // Time to wait for the Node server to respond to the Engine Proxy.
```
Comment (zionts) [inline README.md]: Unrelated, but we should add a note to Heroku customers here to set this to 29s (we should definitely do that in the Engine docs, but here would be nice too)

File: src/index.ts [hunk 3]
```diff
@@ -244,70 +245,82 @@ export class Engine extends EventEmitter {
...
-      if (this.useConfigPrecisely) {
-        throw new Error(
-          `Cannot run Apollo Engine with no frontend. Either specify ` +
-            `at least one frontend in your engine-config or set useConfigPrecisely ` +
-            `to false.`,
-        );
+    if (this.useConfigPrecisely) {
+      // If engineConfig is provided to us as an object rather than a filename,
+      // validate that it contains the two required fields, to give a better
+      // error message than engineproxy's which indicates that you can remove
```
Comment (zionts) [inline src/index.ts]: The ending bit has ambiguous reference (which indicates that you can ...) to me

File: README.md (hunk 2 above)
Comment (glasser) [reply README.md]: Why?

Comment (zionts) [reply README.md]: Idan mentioned that without that people get lots of 503s

Comment (glasser) [reply README.md]: OOOOOOPS I totally misread what line this was on. Sorry.

**Pull request merged**